src/plugins/ui_stateful/edition.ts

```diff
@@ -591,7 +591,7 @@ export class EditionPlugin extends UIPlugin {
     this.selectionEnd = this.selectionStart + previousRefToken!.value.length;
 
     const newRange = this.getters.getRangeFromZone(activeSheetId, newZone);
-    const newRef = this.getRangeReference(newRange, previousRange.parts);
+    const newRef = this.getRangeReference(newRange, [...previousRange.parts]);
```

Comment:
its a typing issue that I kind of tried to avoid. PreviousRnage.parts is readonly,  and it feels weird to pass a readonly rangePart to getRangeReference, 
If you have a good suggestion that does not imply recasting types ^^